src/ClassDiscovery.php

```diff
@@ -120,9 +121,9 @@ public static function setStrategies(array $strategies)
     /**
      * Append a strategy at the end of the strategy queue.
      *
-     * @param DiscoveryStrategy $strategy
+     * @param string $strategy Fully qualified class name to a DiscoveryStrategy
      */
-    public static function appendStrategy(DiscoveryStrategy $strategy)
+    public static function appendStrategy($strategy)
```

Comment:
maybe like this:

```php
ClassDiscovery::setStrategies(['Http\Discovery\MockDiscovery']);
...
class MockDiscovery implements Discovery
{
    ...
}
class MockClient implements Client
{
    public function ... () {} // do nothing
}
```

but indeed with passing strings, you would need quite a few mock classes.